### PR TITLE
TST: Add CodeQL security check

### PR DESCRIPTION
**Pull request by pllim**
@larrybradley and Lee Quick asked me to do this.

Solution | Run time (s) | Problems found
--- | --- | --- |
`bandit` | 20 | 0
CodeQL (JavaScript) | 60 | 0
CodeQL (Python) | 1800 | 0

CodeQL results: https://github.com/spacetelescope/jdaviz/security/code-scanning?query=ref%3Arefs%2Fpull%2F554%2Fmerge+tool%3ACodeQL

### CODE REVIEW

**Comment by pllim** [comment]
Windows failure is unrelated.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
+  push:
```
Comment (pllim) [inline .github/workflows/codeql-analysis.yml:9]: We can consider running this only as weekly cron job or something. 🤷

**Review by ibusko — APPROVED** [review]
I'm no expert on this, but since it can be run with no issues, I approve.

**Review by rosteen — APPROVED** [review]
Weekly seems reasonable

**Comment by pllim** [comment]
I skipped CI because it is unnecessary for the second commit. I guess will need @duytnguyendtn to merge due to branch protection rules. Thanks!

**Pull request merged**